make-docs.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/sh
+
+# SPDX-FileCopyrightText: 2023 SAP SE
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This file is part of FEDEM - https://openfedem.org
+
+print_help () {
+    echo "make-docs.sh"
+    echo ""
+    echo "build the end user documentation for FEDEM."
+    echo ""
+    echo "Usage:"
+    echo "make-docs.sh --option <argument>"
+    echo ""
+    echo "-t, --theory   Build the theory guideline (requires pdflatex)"
+    echo "-u, --user     Build the user guideline (requires pdflatex)"
+    echo "-r, --ref      Build the reference guide (not implemented yet)"
+    echo "-i <folder>, --install <folder>"
+    echo "               Install the produced documents on <folder>"
+    echo "-h, --help     Show this help page"
+}
+
+BUILD_THEORY=false
+BUILD_USER=false
+BUILD_REF=false
+INSTALL=false
+INSTALL_DIR=""
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    -t|--theory)
+      BUILD_THEORY=true
+      shift # past argument
+      ;;
+    -u|--user)
+      BUILD_USER=true
+      shift # past argument
+      ;;
+    -r|--ref)
+      BUILD_REF=true
+      shift # past argument
+      ;;
+    -i|--install)
+      INSTALL=true
+      INSTALL_DIR="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    -h|--help)
+      print_help
+      shift # past argument
+      ;;
+    -*|--*)
+      echo "Unknown option $1"
+      exit 2
+      ;;
+    *)
+  esac
+done
+
+# Print an error if reference guide is asked (to be fixed)
+if [ "$BUILD_REF" = true ] ; then
+    echo "Reference guide building is not implemented on Linux yet!"
+    echo "Please, execute the command again without the option -r, --ref"
+    exit 2
+fi
+
+# Check that something has to be done
+if [ "$BUILD_THEORY" = false ] && [ "$BUILD_USER" = false ] ; then
+    echo "Nothing to be done..."
+    echo "Please, type 'make-docs.sh -h' to see the program usage"
+    exit 0
+fi
+
+# Check that pdflatex is available to build theory or user guides
+if [ "$BUILD_THEORY" = true ] || [ "$BUILD_USER" = true ] ; then
+    pdflatex -v > /dev/null
+    if [ $? -ne 0 ]; then
+        echo "pdflatex cannot be found"
+        echo "Please, install pdflatex. If you already have pdflatex installed,"
+        echo "please set the PATH environment variable to point to it typing"
+        echo "export PATH=$PATH:$PDFLATEX_PATH, with $PDFLATEX_PATH the path where"
+        echo "pdflatex can be found"
+        exit 127
+    fi
+fi
+
+# Time to stop execution if something fails
+set -e
+
+# Save the working folder
+cwd=$(pwd)
+
+if [ "$BUILD_THEORY" = true ] ; then
+    cd $cwd/src/TheoryGuide
+    echo "Generate the Fedem Theory Guide (pdf^)"
```

Comment:
Can remove the "^" character here. It is needed in the BAT-file only where it escapes the following ")".

Reply:
Done in [51dabf6](https://github.com/SAP/fedem-docs/pull/1/commits/51dabf690957ba2743274ab675cbf108e9bc877b)